### PR TITLE
refactor: Remove Spray Json from files refactored during DEV-1627

### PR DESCRIPTION
**Pull request by siers**
### Task Description/Number

Issue Number: DEV-

### PR Type

- [ ] build/chore: maintenance tasks (no production code change)
- [ ] docs: documentation changes (no production code change)
- [ ] feat: represents new features
- [ ] fix: represents bug fixes
- [ ] perf: performance improvements
- [x] refactor: represents production code refactoring
- [ ] test: adding or refactoring tests (no production code change)
- [ ] deprecated: Deprecation warning (ideally referencing a migration guide)

### Basic requirements for bug fixes and features

- [ ] Tests for the changes have been added
- [ ] Docs have been added / updated

### Does this PR introduce a breaking change?

- [ ] Yes

### CODE REVIEW

**Review by BalduinLandolt — APPROVED** [review]
👍

File: webapi/src/main/scala/org/knora/webapi/messages/store/sipimessages/SipiMessages.scala [hunk 1]
```diff
@@ -90,26 +88,31 @@ case class SipiUploadWithoutProcessingResponseEntry(
  *
  * @param uploadedFiles the information about each file that was uploaded.
  */
-case class SipiUploadResponse(uploadedFiles: Seq[SipiUploadResponseEntry])
+case class SipiUploadResponse(uploadedFiles: List[SipiUploadResponseEntry])
+
+object SipiUploadResponseJsonProtocol {
+  implicit val encoderEntry: JsonEncoder[SipiUploadResponseEntry] = DeriveJsonEncoder.gen[SipiUploadResponseEntry]
+  implicit val decoderEntry: JsonDecoder[SipiUploadResponseEntry] = DeriveJsonDecoder.gen[SipiUploadResponseEntry]
```
[the remark this answers is not in the document]
Comment (siers) [reply webapi/src/main/scala/org/knora/webapi/messages/store/sipimessages/SipiMessages.scala:95]: I am in the process of rewriting the last codec during whose refactoring I discovered the codec yeah, will change.

File: webapi/src/main/scala/org/knora/webapi/store/iiif/impl/SipiServiceLive.scala [hunk 2]
```diff
@@ -104,12 +104,10 @@ final case class SipiServiceLive(
...
-          Map(
-            "permission" -> JsString("StoreFile"),
-            "filename"   -> JsString(moveTemporaryFileToPermanentStorageRequestV2.internalFilename),
-            "prefix"     -> JsString(moveTemporaryFileToPermanentStorageRequestV2.prefix),
-          ),
+        "knora-data" -> Json.Obj(
+          "permission" -> Json.Str("StoreFile"),
+          "filename"   -> Json.Str(moveTemporaryFileToPermanentStorageRequestV2.internalFilename),
+          "prefix"     -> Json.Str(moveTemporaryFileToPermanentStorageRequestV2.prefix),
         ),
```
[the remark this answers is not in the document]
Comment (siers) [reply webapi/src/main/scala/org/knora/webapi/store/iiif/impl/SipiServiceLive.scala:111]: ✅ yes, noted

File: webapi/src/main/scala/org/knora/webapi/messages/store/sipimessages/SipiMessages.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (siers) [reply webapi/src/main/scala/org/knora/webapi/messages/store/sipimessages/SipiMessages.scala]: Check it out, TriplestoreService now is also without spray and I got rid of a bunch of verbose lines.

[the remark this answers is not in the document]
Comment (siers) [reply webapi/src/main/scala/org/knora/webapi/messages/store/sipimessages/SipiMessages.scala]: Hopefully the tests pass, in other news, closing this comment thread as it is done.

File: webapi/src/main/scala/org/knora/webapi/store/triplestore/impl/TriplestoreServiceLive.scala [hunk 3]
```diff
@@ -163,7 +162,14 @@ case class TriplestoreServiceLive(
...
       _            <- ZIO.logDebug(s"sparqlHttpAsk - resultString: $resultString")
-      result       <- ZIO.attemptBlocking(resultString.parseJson.asJsObject.getFields("boolean").head.convertTo[Boolean])
+      result <-
+        ZIO
+          .fromEither(for {
+            obj     <- resultString.fromJson[Json.Obj]
+            boolean <- obj.get("boolean").toRight("Missing \"boolean\" key.")
+            result  <- boolean.asBoolean.toRight("Boolean value expected.")
+          } yield result)
+          .mapError(Throwable(_))
```
Comment (BalduinLandolt) [inline webapi/src/main/scala/org/knora/webapi/store/triplestore/impl/TriplestoreServiceLive.scala:172]: you could probably make this a one-liner (or at least almost) by using a JsonCursor. (I have a PR open which would add documentation on that https://github.com/zio/zio-json/pull/1098)

File: webapi/src/main/scala/org/knora/webapi/messages/store/sipimessages/SipiMessages.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (BalduinLandolt) [reply webapi/src/main/scala/org/knora/webapi/messages/store/sipimessages/SipiMessages.scala]: very nice! 💪   
you may want to check out JsonCursor (see comment below) to get rid of even more verbosity

File: webapi/src/main/scala/org/knora/webapi/store/triplestore/impl/TriplestoreServiceLive.scala (hunk 3 above)
Comment (siers) [reply webapi/src/main/scala/org/knora/webapi/store/triplestore/impl/TriplestoreServiceLive.scala:172]: I hope it's fine if I leave this be.

Comment (BalduinLandolt) [reply webapi/src/main/scala/org/knora/webapi/store/triplestore/impl/TriplestoreServiceLive.scala:172]: sure

**Comment by siers** [comment]
@BalduinLandolt I made another useless case class, because the original one creates exceptions and the codec doesn't anticipate that, so I convert to the safe one, filter (like pre-refactor codec did), then convert to the non-safe one

**Comment by BalduinLandolt** [comment]
> I made another useless case class, because the original one creates exceptions and the codec doesn't anticipate that, so I convert to the safe one, filter (like pre-refactor codec did), then convert to the non-safe one

fine by me :)

**Pull request merged**